### PR TITLE
Added buy album feature

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary

Added buy album feature

## Trello Card

https://trello.com/c/U82ij0r7/25-comprar-album-async-await

### CODE REVIEW

File: app/services/albums.js [hunk 1]
```diff
@@ -1,13 +1,11 @@
 const rp = require('request-promise');
 const errors = require('../errors');
 const config = require('../../config/index');
+const models = require('../models');
 
 const base_uri = config.common.api.jsonPlaceholderBaseUri;
 
-const options = {
-  headers: { 'User-Agent': 'Request-Promise' },
-  json: 'true'
```
Comment (arinaldi118) [inline app/services/albums.js]: why did you delete this?

File: app/services/albums.js [hunk 2]
```diff
@@ -25,7 +23,33 @@ const listAlbumPhotos = albumId => {
...
+  models.albums.create(albumData).catch(() => {
+    throw errors.databaseError('Could not create album');
+  });
+
+const getInfoById = albumId => {
+  options.uri = `${base_uri}/albums/${albumId}`;
+  return rp(options)
+    .then(result => {
+      const albumData = JSON.parse(result);
+      delete albumData.userId;
```
Comment (arinaldi118) [inline app/services/albums.js]: Why did you delete this?

File: migrations/migrations/20191120170031-create-user-albums.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+module.exports = {
+  up: (queryInterface, Sequelize) =>
```
Comment (arinaldi118) [inline migrations/migrations/20191120170031-create-user-albums.js]: user_id and album_id must be primary keys

**Review by arinaldi118 — CHANGES_REQUESTED** [review]

File: app/services/albums.js (hunk 2 above)
Comment (tagonzalez) [reply app/services/albums.js]: When requesting for an album in the API, it comes back with a `userId` that's completely unrelated to our DBs user ids. Therefore I found it appropriate to delete it since that information is of no use.

File: migrations/migrations/20191120170031-create-user-albums.js (hunk 3 above)
Comment (tagonzalez) [reply migrations/migrations/20191120170031-create-user-albums.js]: Changed in latest commit!

File: app/services/user_albums.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const models = require('../models');
+const errors = require('../errors');
+
+const albumBoughtByUser = (albumId, userId) =>
+  models.userAlbums
+    .findOne({ where: { albumId, userId } })
+    .then(foundUser => {
+      if (foundUser) {
```
[the remark this answers is not in the document]
Comment (tagonzalez) [reply app/services/user_albums.js]: Changed to a findUserAlbumFunction in latest commit!

File: app/services/albums.js (hunk 1 above)
Comment (tagonzalez) [reply app/services/albums.js]: External API doesn't like these options when I do tests in Postman, I had to remove them to get them to work for the external API's `/albums/:id` endpoint

File: migrations/migrations/20191120170031-create-user-albums.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    queryInterface
+      .createTable('user_albums', {
+        user_id: {
+          type: Sequelize.INTEGER,
+          allowNull: false
+        },
+        album_id: {
+          type: Sequelize.INTEGER,
+          allowNull: false
+        },
```
Comment (tagonzalez) [inline migrations/migrations/20191120170031-create-user-albums.js]: Add foreign keys with 
```
references: {
  model: 'MODEL_NAME',
  key: 'MODEL_PRIMARY_KEY'
}
```

**Review by arinaldi118 — APPROVED** [review]
